### PR TITLE
enhancing this shit

### PR DESCRIPTION
**Pull request by l4zs**
Basically making use of the paper api and replacing deprecated spigot methods.
Also enhancing switches and if statements and refactoring code.

### CODE REVIEW

**Comment by louboecker** [comment]
Have fun going through that all Jilence pepeLaugh

File: src/main/java/de/jilence/jutils/challenges/NoBlockBreak.java [hunk 1]
```diff
@@ -17,11 +17,11 @@
 
 public class NoBlockBreak extends Challenge implements Listener {
 
-    private static NoBlockBreak noBlockBreak = new NoBlockBreak();
+    private static NoBlockBreak noBlockBreak;
 
     @Override
     public void onEnable() {
-
+        noBlockBreak = new NoBlockBreak();
```
Comment (l4zs) [inline src/main/java/de/jilence/jutils/challenges/NoBlockBreak.java:24]: instantiate not initiate xD

File: src/main/java/de/jilence/jutils/utils/ItemBuilder.java [hunk 2]
```diff
@@ -10,27 +10,30 @@
...
 
 public class ItemBuilder {
 
     private ItemStack item;
     private ItemMeta meta;
-    private Material material = Material.STONE;
+    private Material material;
     private int amount = 1;
-    private MaterialData data;
+    private @Nullable MaterialData data;
```
Comment (l4zs) [inline src/main/java/de/jilence/jutils/utils/ItemBuilder.java:32]: MaterialData is deprecated btw,
make sure to switch in the near future.

**Comment by l4zs** [comment]
and btw please compile with a more recent java version, you could simplify so much more things with more recent java versions and java 8 is old af

File: src/main/java/de/jilence/jutils/utils/ItemBuilder.java (hunk 2 above)
Comment (l4zs) [inline src/main/java/de/jilence/jutils/utils/ItemBuilder.java:32]: btw MaterialData is deprecated,
make sure to switch in the near future.

**Comment by warriorzz** [comment]
And maybe move the 'challenges' package to 'challenge.impl', otherwise it is a bit confusing.

**Comment by l4zs** [comment]
was thinking about that too tho but didn't want to do that right away

**Pull request merged**